src/components/SystemMessageList/SystemMessageList.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {useState} from 'react';
+import {SystemMessageNotification, SystemMessageNotificationTypes} from '../SystemMessageNotification/SystemMessageNotification';
+import {getSystemMessages} from '../../api';
+import {useMount} from '../../hooks/useMount';
+
+
+export interface SystemMessageList {
+  id: number;
+  type: SystemMessageNotificationTypes;
+  description: string;
+
+}
+
+// Create function for getting and processing id to local storage
+
+const SystemMessagesList  = () => {
+
+  const [systemMessageList, setSystemMessageList] = useState<SystemMessageList[]>([]);
+
+  const [count, setCount] = useState( 0);
+
+  useMount(async () => {
+      const messages = await getSystemMessages();
+      setSystemMessageList(messages);
+
+  });
+
+     return (
+      <div>
+        {systemMessageList[count] ? <SystemMessageNotification onClick={() => setCount(count+1)} id={systemMessageList[count].id} type={systemMessageList[count].type} description={systemMessageList[count].description} />
```

Comment:
Here is where a big part of your approach will need to change. We have a list of system messages, and we have a list of IDs for messages that have already been dismissed. We need to iterate over those messages, and only display the ones that haven't been dismissed already. You can accomplish this with the `map()` method, which you can read about [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

```
{systemMessageList.map((m: SystemMessageList) => (
  !dismissedMessages.includes(m.id.toString()) ? <SystemMessageNotification onClick={() => setDismissedMessages([...dismissedMessages, m.id.toString()])} id={m.id} type={m.type} description={m.description} />
))}
```

This will display any messages that don't have an ID included in the `dismissedMessages` state, and when the message is clicked it will add the clicked message's ID to the state.

Reply:
@johnpcooke94 I tried this but I am receiving an  Expected an assignment or function call and instead saw an expression error. Have you had issues with something like this before? I'll commit my changes so far.

Reply:
You might need to wrap this ternary in curly braces to signify that it's a javascript object. So like this:

```{!dismissedMessages.includes(m.id.toString()) ? <SystemMessageNotification onClick={() => setDismissedMessages([...dismissedMessages, m.id.toString()])} id={m.id} type={m.type} description={m.description} />
    : null}```